Lib/unittest/mock.py

```diff
@@ -204,6 +204,33 @@ def checksig(*args, **kwargs):
     _setup_func(funcopy, mock, sig)
     return funcopy
 
+def _set_async_signature(mock, original, instance=False, is_async_mock=False):
+    # creates an async function with signature (*args, **kwargs) that delegates to a
+    # mock. It still does signature checking by calling a lambda with the same
+    # signature as the original.
+
+    skipfirst = isinstance(original, type)
+    result = _get_signature_object(original, instance, skipfirst)
+    if result is None:
+        return mock
+    func, sig = result
+    def checksig(*args, **kwargs):
+        sig.bind(*args, **kwargs)
+    _copy_func_details(func, checksig)
+
+    name = original.__name__
+    if not name.isidentifier():
+        name = 'funcopy'
```

Comment:
@graingert - same with this branch, see: 
https://app.circleci.com/pipelines/github/testing-cabal/mock/1428/workflows/7b4dab2f-46de-4dd7-bb8b-5d2000975547/jobs/20351

(and https://github.com/testing-cabal/mock/pull/510)

Reply:
https://github.com/python/cpython/pull/106617